File: svelt/modbus/svelte-test_io/test_remote_server_actions.js
const axios = require('axios');

const SERVER_URL = 'http://127.0.0.1:5000/run-command';

// Test the GET endpoint
async function testGetCommand() {
    try {
        const params = {
            ip: '127.0.0.1',
            // ... Add other parameters if needed
        };

        const response = await axios.get(SERVER_URL, { params });
        console.log('GET Response:', response.data);
    } catch (error) {
        console.error('Error on GET:', error.response.data);
    }
}

// Test the POST endpoint
async function testPostCommand() {
    try {
        const actions = [
            {id: "1", delay: 0, ip: "127.0.0.1", func: "somefunc", args: "someargs"},
            {id: "2", delay: 0.2, ip: "127.0.0.1", func: "somefunc", args: "someargs"},
            // ... Add more actions if needed
        ];

        const response = await axios.post(SERVER_URL, { actions });
        console.log('POST Response:', response.data);
    } catch (error) {
        console.error('Error on POST:', error.response.data);
    }
}

async function runTests() {
    await testGetCommand();
    await testPostCommand();
}

runTests();

File: web_tools/post_relay/test_remote_server_actions.py
import requests

def test_run_command():
    url = "http://127.0.0.1:5000/run-command"
    # docker will direct this to the docker node attached to port 5000
    # python_server_dest must be running on that docker image and on the 172.17.0.2 image 
    params = {
        "route": "172.17.0.2",
        "ip": "172.17.0.3",
        "port": "8080",
        "action": "sample_action",
        "deviceId": "sample_device",
        "type": "sample_type",
        "offset": "sample_offset",
        "value": "sample_value",
        "rowId": "sample_rowId"
    }
    
    try:
        response = requests.get(url, params=params)
        response.raise_for_status()
        
        print("Server Response:", response.json())
    except requests.RequestException as error:
        print(f"Error testing server: {error}")
        print("Server Response:", response.json())

# Test the POST endpoint
def test_post_command():
    actions = [
        {"id": "1", "delay": 0, "ip": "127.0.0.1", "func": "somefunc", "args": "someargs"},
        {"id": "2", "delay": 0.2, "ip": "127.0.0.1", "func": "somefunc", "args": "someargs"},
        # ... Add more actions if needed
    ]

    headers = {
        'Content-Type': 'application/json'
    }

    response = requests.post(SERVER_URL, data=json.dumps({'actions': actions}), headers=headers)
    print("POST Response:", response.json())
    
if __name__ == "__main__":
    test_run_command()